pyct/report.py

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env python
+# Import and print the library version and filesystem location for each Python package or shell command specified
+#
+# bash usage: $ report numpy pandas python conda
+# python usage: >>> from report import report; report("numpy","pandas","python","conda")
+
+from __future__ import print_function
+import os.path, importlib, subprocess, platform, sys
+
+def report(*packages):
+    """Import and print location and version information for specified Python packages"""
+    accepted_commands = ['python','conda']
+    for package in packages:
+        loc = "not installed in this environment"
+        ver = "unknown"
+
+        try:
+            module = importlib.import_module(package)
+            loc =  os.path.dirname(module.__file__)
+
+            try:
+                ver = str(module.__version__)
+            except Exception:
+                pass
+        
+        except (ImportError, ModuleNotFoundError):
+            if package in accepted_commands:
+                try:
+                    # See if there is a command by that name and check its --version if so
+                    try:
+                        loc = subprocess.check_output(['which','-a',      package]).decode().splitlines()[0].strip()
```

Comment:
I think the upshot from the articles that Chris links to is that we should use 'command','-v' here.  I tested it and it seems to work the same as `which`, while promising to work properly in more cases, so I think we should switch to it.  I think you said that `shell=True` was causing problems, but I'm not seeing any difference when I add that to either `which` or `command`; can you document what happens when it is used?

Reply:
When the command is run with `shell=True` included in that line it still appears to work for `system` and python packages, but for `conda` or `python` it gives back 
"conda=unknown                  # not installed in this environment"

Reply:
Bizarre!  I tried it with conda and it worked fine for me, but with python it doesn't. Ok, no `shell=True`!